python_modules/dagster/dagster/_core/executor/multiprocess.py

```diff
@@ -290,6 +293,8 @@ def execute(
                     for key in empty_iters:
                         del active_iters[key]
                         del term_events[key]
+                        if key in processes:
+                            del processes[key]
```

Comment:
this is the impactful part, the rest were just during process cleanup at the end of the run